asm/variable/u64.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package variable // import "go.opentelemetry.io/ebpf-profiler/asm/variable"
+import "sort"
+
+type U64 interface {
```

Comment:
The `U64` is slightly confusing as it can be anything. Perhaps `Expression` (since some implementations nest other implementations of this interface) or something similar generic enough?

Reply:
I was choosing between BV64 and U64 (similar to how they are called in s3 solver).
Renamed to `Expression`